### PR TITLE
kubeadm: ensure proper parsing of SSR username

### LINKED ISSUE #3118: [Failing Test] kubeadm-kinder-upgrade-1-31-latest
https://testgrid.k8s.io/sig-release-master-informing#kubeadm-kinder-upgrade-1-31-latest

```
I1101 04:57:03.517982    3765 round_trippers.go:560] GET https://172.17.0.7:6443/api/v1/nodes/system:node:kinder-upgrade-control-plane-1?timeout=10s 404 Not Found in 3 milliseconds
the ConfigMap "kubeadm-config" in the "kube-system" namespace was not found
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.enforceRequirements
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/common.go:102
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.runPlan
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/plan.go:104
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.newCmdPlan.func1
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/plan.go:78
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.1/command.go:985
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.1/command.go:1117
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.1/command.go:1041
k8s.io/kubernetes/cmd/kubeadm/app.Run
	k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:47
main.main
	k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:25
runtime.main
	runtime/proc.go:272
runtime.goexit
	runtime/asm_amd64.s:1700
[upgrade/init config] FATAL
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.enforceRequirements
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/common.go:104
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.runPlan
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/plan.go:104
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.newCmdPlan.func1
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/plan.go:78
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.1/command.go:985
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.1/command.go:1117
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.1/command.go:1041
k8s.io/kubernetes/cmd/kubeadm/app.Run
	k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:47
main.main
	k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:25
runtime.main
	runtime/proc.go:272
runtime.goexit
	runtime/asm_amd64.s:1700
Error: failed to exec action kubeadm-upgrade: exit status 1
 exit status 1
```

### PR DESCRIPTION
**Pull request by neolit123**
#### What this PR does / why we need it:

- Split the code that tries to get node name from SSR into a new function getNodeNameFromSSR(). Unit test the function.
- Fix error that the "system:nodes:" prefix was not trimmed.
- Fix misleading errors around FetchInitConfigurationFromCluster. This function performs multiple actions, and the "get node" action can also be of type apierrors.NotFound(). This creates confusion in the returned error in enforceRequirement during upgrade. Fix this problem.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubeadm/issues/3118

folowup:
- https://github.com/kubernetes/kubernetes/pull/128457

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/upgrade/common.go [hunk 1]
```diff
@@ -96,11 +94,6 @@ func enforceRequirements(flagSet *pflag.FlagSet, flags *applyPlanFlags, args []s
 
 	initCfg, err := configutil.FetchInitConfigurationFromCluster(client, printer, "upgrade/config", false, false)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			_, _ = printer.Printf("[upgrade/config] In order to upgrade, a ConfigMap called %q in the %q namespace must exist.\n", constants.KubeadmConfigConfigMap, metav1.NamespaceSystem)
-			_, _ = printer.Printf("[upgrade/config] Use 'kubeadm init phase upload-config --config your-config.yaml' to re-upload it.\n")
-			err = errors.Errorf("the ConfigMap %q in the %q namespace was not found", constants.KubeadmConfigConfigMap, metav1.NamespaceSystem)
-		}
 		return nil, nil, nil, nil, errors.Wrap(err, "[upgrade/init config] FATAL")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/upgrade/common.go:97]: just return the underlying error. there is already printer output that it's trying to get the CM.

**Review by neolit123 — COMMENTED** [review]
/kind bug
/priority critical-urgent

/cc @SataQiu @pacoxu

File: cmd/kubeadm/app/util/config/cluster.go [hunk 2]
```diff
@@ -231,6 +230,22 @@ func getNodeNameFromKubeletConfig(fileName string) (string, error) {
...
+// getNodeNameFromSSR reads the node name from the SelfSubjectReview for a given client.
+// If the kubelet.conf is passed as fileName it can be used to retrieve the node name.
+func getNodeNameFromSSR(client clientset.Interface) (string, error) {
+	ssr, err := client.AuthenticationV1().SelfSubjectReviews().
+		Create(context.Background(), &authv1.SelfSubjectReview{}, metav1.CreateOptions{})
+	if err != nil {
+		return "", err
+	}
+	user := ssr.Status.UserInfo.Username
+	if !strings.HasPrefix(user, constants.NodesUserPrefix) {
```
Comment (pacoxu) [inline cmd/kubeadm/app/util/config/cluster.go:242]: The key of this fix  is this line.

**Review by pacoxu — COMMENTED** [review]
/lgtm
/hold 
for others to take a look if others have time.

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Comment by neolit123** [comment]
in an ideal world we should have presubmit test jobs for kubeadm upgrade like the one we have in the kubeadm repo.
it does upgrade for latest -> latest (same version) just to check if some parts of that failed.

**Pull request merged**